File: src/main/java/com/example/service/NotificationService.java
package com.example.service;

import java.util.List;

import com.example.dto.NotificationDTO;

public interface NotificationService {

	/**
	 * Retourne la liste des notifications d'un utilisateur donné
	 * @param userId id de l'utilisateur
	 * @return la liste des notifications d'un utilisateur donné
	 */
	List<NotificationDTO> getNotificationByUserId(Long userId);
	
	/**
	 * Marque toutes les notifications de l'utilisateur comme lues.
	 * @param userId l'id de l'utilisateur
	 */
	void readAllNotification(Long userId);
	
	/**
	 * Crée une notifications
	 * @param notif données de la notification
	 * @return DTO de la notification créée
	 */
	NotificationDTO createNotification(NotificationDTO notif);
	
 
}

File: src/main/resources/static/js/notification/NotificationService.js
/**
 * Created by Morgane TROYSI on 22/04/2017.
 */
(function() {
    'use strict';

    var helloApp = angular.module('hello');

    /** @ngInject */
    helloApp.service('NotificationService', function(NotificationWS, $injector) {
        var svc = {};

        /**
         * Retourne les notifications d'un utilisateur
         * @param id l'id de l'utilisateur
         */
        svc.getNotificationByUser = function(id) {
            return NotificationWS.getNotificationByUser(id).then(function(response) {
                return response.data;
            });
        };

        /**
         * Crée une notification
         * @param notification données de la notif à créer
         */
        svc.createNotification = function(notification) {
            return NotificationWS.createNotification(notification).then(function(response) {
                return response.data;
            });
        };

        /**
         * Marque toutes les notifications de l'utilisateur comme lues
         * @param idUser id de l'utilisateur
         */
        svc.readAllNotification = function(idUser) {
            var list = $injector.get('CommonNotificationService').getListNotification() || [];
            list.forEach(function(notif) {
                notif.isRead = true;
            });
            return NotificationWS.readAllNotification(idUser);
        };

        return svc;
    })
})();